typesafety/test_primitives/test_tracing/test_collect_traces.yml

```diff
@@ -0,0 +1,13 @@
+- case: collect_traces_context_manager_return_type_hint
+  disable_cache: true
+  main: |
+    from returns.primitives.tracing import collect_traces
+
+    reveal_type(collect_traces)  # N: Revealed type is 'def () -> contextlib._GeneratorContextManager[None]'
+
+- case: collect_traces_context_manager_return_type
+  disable_cache: true
+  main: |
+    from returns.primitives.tracing import collect_traces
+
+    reveal_type(collect_traces())  # N: Revealed type is 'contextlib._GeneratorContextManager[None]'
```

Comment:
Can we use `@overload` to fix this type?

Reply:
I've tried without success!

Implementation:
```python
@overload
def collect_traces() -> Iterator[None]:
    ...

@contextmanager
def collect_traces():  # I've tried with `def collect_traces() -> Iterator[None]` as well
    # IMPLEMENTATION
```

Errors:
```sh
/returns/returns/primitives/tracing:9: error: Single overload definition, multiple required
/returns/returns/primitives/tracing:14: error: Overloaded function implementation cannot produce return type of signature 1
```

Reply:
I've found a possible answer, https://github.com/python/mypy/issues/1317#issuecomment-207564973

```python
def collect_traces() -> Iterator[None]:
    # IMPLEMENTATION

collect_traces = contextmanager(collect_traces)  # type: ignore
```

Output of `reveal_type`:
```sh
Revealed type is 'def () -> typing.Iterator[None]'
```

I didn't like this resolution so much but it works!
What do you think?

Reply:
The above solution fixes #443, as well!

Reply:
Awesome! 👍

Reply:
I was wrong about this:
> The above solution fixes #443, as well!

The solution cannot fixes, see the errors below:
```sh
E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output: 
E   Expected:
E     main:7: note: Revealed type is 'def () -> builtins.int' (diff)
E   Actual:
E     main:3: error: Too many arguments for "collect_traces" (diff)
E     main:7: note: Revealed type is 'typing.Iterator[None]' (diff)
E   
E   Alignment of first line difference:
E     E: main:7: note: Revealed type is 'def () -> builtins.int'
E     A: main:3: error: Too many arguments for "collect_traces"
```

But fixes the current issue about return type when used as a context manager!